File: 3068-find-the-maximum-sum-of-node-values/3068-find-the-maximum-sum-of-node-values.cpp
class Solution {
public:
    long long maximumValueSum(vector<int>& nums, int k, vector<vector<int>>& edges) {
        int n = nums.size();
        vector<int> v;
        long long sum = 0;
        
        
        for(auto x : nums){
            sum+=x;
            v.push_back((x^k)-x);
        }
        
        sort(v.rbegin(), v.rend());
        
        for(int i=0;i<n-1;i+=2){
            if(v[i]+v[i+1] > 0){
                sum+=(v[i]+v[i+1]);
            }
        }
        
        return sum;
        
    }
};

File: 3068-find-the-maximum-sum-of-node-values/3068-find-the-maximum-sum-of-node-values.py
class Solution:
    def maximumValueSum(self, nums: List[int], k: int, 
                              edges: List[List[int]]) -> int:

        dif = sorted(map(lambda x: x - (x^k), nums)) 
        
        return sum(nums) - sum(filter(lambda x: x < 0, 
                         map(sum, zip(dif[::2],dif[1::2]))))